StoreTestWPF.ViewModel/ViewModels/StoreViewModel.cs

```diff
@@ -0,0 +1,206 @@
+﻿using StoreTestWPF.DAL;
+using StoreTestWPF.DAL.Models;
+using StoreTestWPF.ViewModel.Interfaces;
+using StoreTestWPF.ViewModel.Utils;
+using System.Collections.ObjectModel;
+using System.Data.Entity;
+using System.Linq;
+using System.Windows.Input;
+
+namespace StoreTestWPF.ViewModel.ViewModels
+{
+    public sealed class StoreViewModel : ViewModelBase
+    {
+        private CakeViewModel selectedCake;
+        private bool isCakeSelected;
+        private ICommand loadCommand;
+        private ICommand addCommand;
+        private ICommand editCommand;
+        private ICommand deleteCommand;
+        private IViewService viewService;
+        private CakeStoreDbContext dbContext;
+
+        public ObservableCollection<CakeViewModel> Cakes { get; private set; }
+
+        public CakeViewModel SelectedCake
+        {
+            get { return this.selectedCake; }
+            set
+            {
+                if (this.selectedCake != value)
+                {
+                    this.selectedCake = value;
+                    OnPropertyChanged(nameof(this.SelectedCake));
+                    OnPropertyChanged(nameof(this.IsCakeSelected));
+                }
+            }
+        }
+
+        public bool IsCakeSelected
+        {
+            get
+            {
+                return !(this.selectedCake == null);
```

Comment:
Setter and getter manage different values.

Reply:
Fixed